internal/components/component.go

```diff
@@ -110,7 +110,7 @@ func ConstructServicePort(current *corev1.ServicePort, port int32) corev1.Servic
 	return corev1.ServicePort{
 		Name:        current.Name,
 		Port:        port,
-		TargetPort:  current.TargetPort,
+		TargetPort:  intstr.FromInt32(port),
```

Comment:
I think we should only set the target port if it was previously set, otherwise do nothing. Some components may want to set a [target port](https://github.com/open-telemetry/opentelemetry-operator/blob/main/internal/components/component.go#L42) others don't